src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataSerializerContext.cs

```diff
@@ -161,6 +163,22 @@ internal IEdmTypeReference GetEdmType(object instance, Type type)
                         typeof(IEdmObject).Name);
                 }
             }
+            else if (typeof(IEnumerable).IsAssignableFrom(type) &&
+                typeof(IEdmObject).IsAssignableFrom(type.GetGenericArguments().FirstOrDefault()))
```

Comment:
Web API recommends to use "EdmEnittyObjectCollection, EdmComplexObjectCollection, EdmEnumObjectCollection" for the collection.

Reply:
my approach is consistent with the approaches used in this class and others in webapi.

---
In reply to: [215081845](https://github.com/OData/WebApi/pull/1607#discussion_r215081845) [](ancestors = 215081845)